src/dbwrapper.cpp

```diff
@@ -158,14 +158,16 @@ struct CDBBatch::WriteBatchImpl {
 
 CDBBatch::CDBBatch(const CDBWrapper& _parent)
     : parent{_parent},
-      m_impl_batch{std::make_unique<CDBBatch::WriteBatchImpl>()} {};
+      m_impl_batch{std::make_unique<CDBBatch::WriteBatchImpl>()}
+{
+    Clear();
```

Comment:
nit: Is the `Clear()` here still necessary, now that `Clear` doesn't do anything but call `Clear` on the internal batch?

Reply:
We're calling [`batch.Clear()`](https://github.com/bitcoin/bitcoin/blob/master/src/txdb.cpp#L133) after [the first batch is written](https://github.com/bitcoin/bitcoin/blob/master/src/txdb.cpp#L130), which does [`rep_.resize(kHeader)`](https://github.com/bitcoin/bitcoin/blob/master/src/leveldb/db/write_batch.cc#L37) (hence the original assert).
For consistency we need this presizing for the very first batch as well.